app/integrations/comp_and_pen/scheduled_message_helpers.py

```diff
@@ -0,0 +1,207 @@
+                    with critical severity, and then re-raised.
+        """
+        if self.dynamodb_table is None:
+            self._connect_to_dynamodb()
+
+        # send messages and update entries in dynamodb table
+        with self.dynamodb_table.batch_writer() as batch:
+            for item in comp_and_pen_messages:
+                participant_id = item.get('participant_id')
+                payment_id = item.get('payment_id')
+
+                item.pop('is_processed', None)
+
+                # update dynamodb entries
+                try:
+                    batch.put_item(Item=item)
+                    current_app.logger.info(
+                        'updated_item from dynamodb ("is_processed" should no longer exist): %s', item
+                    )
+                except Exception as e:
+                    current_app.logger.critical(
+                        'Exception attempting to update item in dynamodb with participant_id: %s and payment_id: %s - '
+                        'exception_type: %s exception_message: %s',
+                        participant_id,
+                        payment_id,
+                        type(e),
+                        e,
+                    )
+
+    def send_comp_and_pen_sms(
+        self,
+        service: Service,
+        template: Template,
+        sms_sender_id: str,
+        comp_and_pen_messages: list[dict],
+        perf_to_number: str,
+    ) -> None:
+        """
+        Sends scheduled SMS notifications to recipients based on the provided parameters.
+
+        Args:
+            :param service (Service): The service used to send the SMS notifications.
+            :param template (Template): The template used for the SMS notifications.
+            :param sms_sender_id (str): The ID of the SMS sender.
+            :param comp_and_pen_messages (list[dict]): A list of dictionaries from the dynamodb table containing the
+                details needed to send the messages.
+            :param perf_to_number (str): The recipient's phone number.
+
+        Raises:
+            Exception: If there is an error while sending the SMS notification.
+        """
+        for item in comp_and_pen_messages:
+            vaprofile_id = str(item.get('vaprofile_id'))
+            participant_id = item.get('participant_id')
+            payment_id = item.get('payment_id')
+            payment_amount = str(item.get('paymentAmount'))
+
+            current_app.logger.info(
+                'sending - item from dynamodb - vaprofile_id: %s | participant_id: %s | payment_id: %s',
+                vaprofile_id,
+                participant_id,
+                payment_id,
+            )
+
+            # Use perf_to_number as the recipient if available. Otherwise, use vaprofile_id as recipient_item.
+            recipient = perf_to_number
+            recipient_item = (
+                None
+                if perf_to_number is not None
```

Comment:
**Request change or confirmation of functionality**
I think perf_to_number will never be `None`. Since on line `154` it is formatted as a string, I believe if paymentAmount does not exist or is None, it will be assigned as `"None"` (None in quotes)

Reply:
Test example 

```
test_item = {} 

def example_method(item): 
  payment_amount = str(item.get('paymentAmount'))
  print("payment_amount is: ", payment_amount)
  	
  return payment_amount is not None
 
is_payment_amount_not_None = example_method(test_item)
 
print(is_payment_amount_not_None)
```

Output 
``` 
('payment_amount is: ', 'None')
True
```

https://pythonsandbox.com/code/pythonsandbox_u101954_Yu8JsgnXZAv5vgz3QyoDlXED_v1.py

Reply:
I don't see where you are talking about `perf_to_number ` being cast as a str()? What's happening is more like this than your shared example. Idk how to do your fancy sandbox thing, but if you throw that in a file and run it, you'll see that it is indeed `None`
```python
import os

item = os.environ.get('ITEM')
print(f'{item=}')

def print_item(item: str):
    print(f'{item=}')
    print(f'{type(item)=}')

print_item(item)

######
# output:
# item=None
# item=None
# type(item)=<class 'NoneType'>
```

Reply:
I'm looking at the wrong variable. I'm dumb this is OK.